src/hierarchies/abstract_hierarchy.h

```diff
@@ -13,63 +13,70 @@
 #include "hierarchy_id.pb.h"
 #include "src/utils/rng.h"
 
-//! Abstract base template class for a hierarchy object.
-
-//! This template class represents a hierarchy object in a generic iterative
-//! BNP algorithm, that is, a single set of unique values with their own prior
-//! distribution attached to it. These values are part of the Markov chain's
-//! state chain (which includes multiple hierarchies) and are simply referred
-//! to as the state of the hierarchy. This object also corresponds to a single
-//! cluster in the algorithm, in the sense that its state is the set of
-//! parameters for the distribution of the data points that belong to it. Since
-//! the prior distribution for the state is often the same across multiple
-//! different hierarchies, the hyperparameters object is accessed via a shared
-//! pointer. Lastly, any hierarchy that inherits from this class contains
-//! multiple ways of updating the state, either via prior or posterior
-//! distributions, and of evaluating the distribution of the data, either its
-//! likelihood (whose parameters are the state) or its marginal distribution.
+//! Abstract base class for a hierarchy object.
+//! This class is the basis for a curiously recurring template pattern (CRTP)
+//! for `Hierarchy` objects, and is solely composed of interface functions for
+//! derived classes to use. For more information about this pattern, as well
+//! the list of methods required for classes in this inheritance tree, please
+//! refer to the README.md file included in this folder.
+
+//! This abstract class represents a Bayesian hierarchical model:
+//! x_1, ..., x_n \sim f(x | \theta)
```

Comment:
I think it's nice to have math formulas here, but they are not displayed correctly in the docs. Also the enumerations don't work as intended

Reply:
I'll hopefully figure it out in the next couple days